Sources/HTTP/PoCSocket/PoCSocketConnectionListener.swift

```diff
@@ -195,10 +195,23 @@ public class PoCSocketConnectionListener: ParserConnecting {
 
     /// Starts reading from the socket and feeding that data to the parser
     public func process() {
-        try! socket?.setBlocking(mode: true)
-
-        let tempReaderSource = DispatchSource.makeReadSource(fileDescriptor: socket?.socketfd ?? -1,
-                                                             queue: socketReaderQueue)
+        let tempReaderSource: DispatchSourceRead
```

Comment:
Not compiled:

```swift
guard let socket = socket else {
    preconditionFailure("Entered \(#function) but socket was nil")
}

do {
    try socket.setBlocking(mode: true)
} catch {
    assertionFailure("Failed to set socket as blocking")
    return
}

let tempReaderSource = DispatchSource.makeReadSource(fileDescriptor: socket.socketfd, 
                                                     queue: socketReaderQueue)

...
```

Reply:
I specifically put a comment in the next line explaining why I don't think a guard is appropriate.

Reply:
Sorry, I don't understand the comment.  How does `guard` encourage anything? It just makes sure the error case is handled clearly.

"Strong" is the wrong terminology for all this anyway, unwrapping an optional does not make it strong in Swift parlance.

Reply:
I'm not sure I understand this either. Can you help me understand how the use of `guard` vs. `if let` affects the likelihood that someone in the future will make a change to use `strongSocket` in the closure, and indeed why having both `strongSocket` and `socket` helps?

Reply:
I still don't have a clear understanding if why this approach is needed, i.e.;
* Why guard is inappropriate
* Why the use of `strongSocket` when `socket` is never a weak reference.

Having said that, I'd rather get the fix in and clean this up later if thats the only way to make progress.